src/Native/Runtime/portable.cpp

```diff
@@ -316,9 +318,10 @@ COOP_PINVOKE_HELPER(Array *, RhpNewArrayAlign8, (EEType * pArrayEEType, int numE
     }
     UInt8* result = acontext->alloc_ptr;
     int requiresAlignObject = ((uint32_t)result) & 7;
-    if (requiresAlignObject) size += 12;
+    size_t paddedSize = size;
+    if (requiresAlignObject) paddedSize += 12;
```

Comment:
This can hit integer overflow and misbehave. There are other places in this method that can misbehave due to integer overflow. It would be nice to at least add TODO about it.

Reply:
What would be the right behaviour here, to throw (OOM/Overflow)?

Reply:
Or shall I just do something like this for `#ifndef HOST_64BIT` 

```
        // Perform the size computation using 64-bit integeres to detect overflow
        uint64_t size64 = (uint64_t)baseSize + ((uint64_t)numElements * (uint64_t)pArrayEEType->get_ComponentSize());
        size64 = (size64 + (sizeof(UIntNative) - 1)) & ~(sizeof(UIntNative) - 1);

        size = (size_t)size64;
        if (size != size64)
        {
            ASSERT_UNCONDITIONALLY("NYI");  // TODO: Throw overflow
        }
```
?

Reply:
Yes, it would be nice to add `ASSERT_UNCONDITIONALLY` TODO for now that will be eventually replaced by throwing proper exception.

Reply:
Think I've got every possible overflow covered now.